crates/cdk/src/test.rs

```diff
@@ -0,0 +1,55 @@
+use std::{
+    fmt::Debug,
+    path::PathBuf,
+    process::{Command, Stdio},
+};
+
+use anyhow::{Result, anyhow};
+use clap::Parser;
+
+use cargo_builder::{package::PackageInfo, cargo::Cargo};
+
+use crate::cmd::PackageCmd;
+
+/// Builds and runs the Connector in the current working directory
+#[derive(Debug, Parser)]
+pub struct TestCmd {
+    #[clap(flatten)]
+    package: PackageCmd,
+
+    #[clap(short, long, value_name = "PATH")]
+    config: PathBuf,
+
+    /// Extra arguments to be passed to cargo
+    #[clap(raw = true)]
+    extra_arguments: Vec<String>,
+}
+
+impl TestCmd {
+    pub(crate) fn process(self) -> Result<()> {
+        let opt = self.package.as_opt();
+        let p = PackageInfo::from_options(&opt)?;
+
+        let cargo = Cargo::build()
+            .profile(opt.release)
+            .lib(false)
+            .package(p.package_name())
+            .extra_arguments(self.extra_arguments)
+            .build()?;
+
+        cargo.run()?;
+
+        let status = Command::new(p.target_bin_path()?)
```

Comment:
this should be using deployer API so we have consistent deployment strategy

Reply:
Done